pkg/services/ngalert/state/manager.go

```diff
@@ -180,7 +181,10 @@ func (st *Manager) setNextState(ctx context.Context, alertRule *ngModels.AlertRu
 	currentState.LastEvaluationString = result.EvaluationString
 	currentState.TrimResults(alertRule)
 	oldState := currentState.State
+	oldReason := currentState.EvaluationReason
 
+	// The underlying reason for a state is always the result of the evaluation.
+	currentState.EvaluationReason = result.State
```

Comment:
I think `StateReason` would be more self-explanatory.

Reply:
Also, I think it will be more explicit if we set reason only when mapping the original state to another state, i.e. Error -> Alerting | OK, NoData -> Alerting | OK

Reply:
I've been thinking about `Reason` vs `Cause`, let me know if you have opinions on which is clearer.

Now that State is renamed to AlertInstance, I can probably make `EvaluationState` back into `State` so it's not so verbose, too.

Reply:
I like both :)